### DISCUSSION TITLE
Single lined object until it reaches max len

### DISCUSSION THREAD

**Discussion by miguelbogota** [eslint/eslint#14389] [Q&A]
Hi ✋🏼

I've been trying to make objects single lined until it reached the max length. I tried to use the `object-property-newline` and also `no-multiple-empty-lines` but none of them worked. The final goal I want to achieve is the following:

With a max length of: 80

```js
/* Bad code since it exceeds the 80 limit */
const object = { propertyName1: 'something', propertyName2: 'something', propertyName3: 'something', propertyName4: 'something' }
```

```js
/* Bad code since it's below the 80 limit */
const object = {
  property1: 'something',
  property2: 'something',
}
```

If the limit hasn't been reached the `--fix` command can fix to have each property in the same line like this:
```js
/* Good code since single lined it's below the 80 limit */
const object = { property1: 'something', property2: 'something' }
```

But once it reaches the limit with the `--fix` command can fix to have each property in a new line like this:
```js
/* Good code since single lined it will exceeds the 80 limit */
const object = {
  propertyName1: 'something',
  propertyName2: 'something',
  propertyName3: 'something',
  propertyName4: 'something',
}
```

The `object-property-newline` and `no-multiple-empty-lines` can help me enforcing always new lines but not enforce single line if the limit hasn't been reached.

If anyone can help I'll be more than thankful!!

**Comment by jooohny** [comment]
No updates?
